### PR TITLE
Add Java Docs

### PR DESCRIPTION
**Pull request by anqi20**

### CODE REVIEW

File: src/main/java/seedu/eduke8/option/OptionList.java [hunk 1]
```diff
@@ -19,11 +22,21 @@ public OptionList(ArrayList<Displayable> options) {
...
         return this.options;
     }
 
+    /**
+     * Finds option with the specific description in the OptionList object.
+     *
+     * @param description Description of option to be found.
+     */
     @Override
     public Displayable find(String description) throws Eduke8Exception {
```
Comment (kstonekuan) [inline src/main/java/seedu/eduke8/option/OptionList.java]: I think you are missing the return here

File: src/main/java/seedu/eduke8/explanation/Explanation.java [hunk 2]
```diff
@@ -11,17 +14,30 @@ public Explanation(String description) {
         wasShown = false;
     }
 
+    /**
+     * Returns the explanation of the question.
+     *
+     * @return explanation description.
```
Comment (Zhi-You) [inline src/main/java/seedu/eduke8/explanation/Explanation.java]: Should the e be a capital letter E here?

File: src/main/java/seedu/eduke8/option/OptionList.java [hunk 3]
```diff
@@ -34,6 +47,11 @@ public Displayable find(String description) throws Eduke8Exception {
         throw new Eduke8Exception(ERROR_OPTION_DOES_NOT_EXIST);
     }
 
+    /**
+     * Returns the index of the correct option in the OptionList object.
+     *
+     * @return Index of the correct option in the OptionList object.
```
Comment (Zhi-You) [inline src/main/java/seedu/eduke8/option/OptionList.java:55]: Think you are missing a @throw here

**Review by Zhi-You — COMMENTED** [review]
Will be good to merge after the changes mentioned.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -297,37 +297,38 @@ The timer would start as soon as the options are being printed out.
...
+This `getCommand()` method takes in the time left on the timer (`userTimer - timePassed`) instead of the timer set by the user. 
+This is because, we want to avoid the situation where the user is able to extend the timer by entering in any random value (except number 1 to 4) as this would cause the timer to reset everytime. 
 Hence, in order to prevent this bug, for this `getCommand()` method, we have to use the time left on the timer instead. 
 The subsequent steps are similar to the one stated above. 
 
-If the `command` is an instance of `AnswerCommand` and `IncompleteCommand`, the loop will end and the `SingleTopicQuiz` object will proceed to execute the `Command` for the last time, using the `execute(optionList, ui)` method.
+The loop will end if the user has written an input or the time is up for that question. 
+The `SingleTopicQuiz` object will then proceed to execute the `Command` for the last time, using the `execute(optionList, ui)` method.
```
Comment (kstonekuan) [inline docs/DeveloperGuide.md:331]: Good additions to the DG

File: src/main/java/seedu/eduke8/option/OptionList.java [hunk 5]
```diff
@@ -19,11 +22,23 @@ public OptionList(ArrayList<Displayable> options) {
...
         return this.options;
     }
 
+    /**
+     * Finds option with the specific description in the OptionList object.
+     *
+     * @param description Description of option to be found.
+     * @return Displayable Option
+     * @throws Eduke8Exception If the option does not exist.
+     */
```
Comment (kstonekuan) [inline src/main/java/seedu/eduke8/option/OptionList.java:41]: Looks good

**Review by kstonekuan — APPROVED** [review]
LGTM

**Pull request merged**